### PR TITLE
fix(resize): Add command to resize xfs volumes

### PR DESCRIPTION
**Pull request by payes**
Signed-off-by: Payes <payes.anand@mayadata.io>

### CODE REVIEW

File: pkg/iscsi/v1alpha1/mount.go [hunk 1]
```diff
@@ -46,7 +47,13 @@ func ResizeVolume(volumePath string) error {
 			if err := util.ReScan(); err != nil {
 				return err
 			}
-			if err := util.ReSizeFS(mpt.Device); err != nil {
+			switch fsType {
+			case "ext4":
+				err = util.ResizeExt4(mpt.Device)
+			case "xfs":
+				err = util.ResizeXFS(volumePath)
```
Comment (mynktl) [inline pkg/iscsi/v1alpha1/mount.go]: What about other types? Should we return an error `not supported or manual op needed`?

File: pkg/iscsi/v1alpha1/mount.go [hunk 2]
```diff
@@ -46,7 +47,13 @@ func ResizeVolume(volumePath string) error {
...
+			}
+			if err != nil {
```
Comment (prateekpandey14) [inline pkg/iscsi/v1alpha1/mount.go]: add default case with error if either of the type not there ?

File: pkg/service/v1alpha1/utils.go [hunk 3]
```diff
@@ -26,9 +26,18 @@ const (
 )
 
 var (
-	ValidFSTypes = []string{FSTypeExt2, FSTypeExt3, FSTypeExt4, FSTypeXfs}
+	ValidFSTypes = []string{FSTypeExt4, FSTypeXfs}
```
Comment (prateekpandey14) [inline pkg/service/v1alpha1/utils.go]: are we don't want to support ext3 ?

File: pkg/service/v1alpha1/controller_utils.go [hunk 4]
```diff
@@ -130,6 +130,12 @@ func (cs *controller) validateVolumeCreateReq(req *csi.CreateVolumeRequest) erro
 			"failed to handle create volume request: missing storage class parameter cas-type",
 		)
 	}
+	if !isValidFStype(req.GetParameters()["fsType"]) {
+		return status.Error(
+			codes.InvalidArgument,
+			"failed to handle create volume request, invalid fsType",
```
Comment (prateekpandey14) [inline pkg/service/v1alpha1/controller_utils.go]: can we have fstype value as well in retun error ?

File: pkg/service/v1alpha1/utils.go (hunk 3 above)
Comment (payes) [reply pkg/service/v1alpha1/utils.go]: Let's support only ext4 and xfs for now, since we have the complete testing suite for just these 2 filesystems.

**Pull request merged**